### PR TITLE
Workspace with unrestricted internet access

### LINKED ISSUE #2482: Workspace with unrestricted internet access
## Description

As a TRE Administrator
I want to deploy a workspace with unrestricted internet access
So that researchers can work on open data using TRE tooling

## Acceptance criteria

- [ ] New template reuses base template terraform assets (maybe pull from git commit)
- [ ] Firewall rule added in pipeline that allows all traffic out of the Workspace address space

### PR DESCRIPTION
**Pull request by marrobi**
# Resolves #2482

## What is being addressed
Create a workspace that allows all traffic to the internet. this is for scenarios where the TRE is to be used for data that does not need to be protected.

## How is this addressed
- Download base template resources in Dockerfile
 - Disable airlock
- Provide provider.tf matching base workspace
- [ ] Add docs

### CODE REVIEW

**Comment by marrobi** [comment]
@tamirkamara @tanya-borisova welcome your feedback on this approach to building on the base workspace. 

Porter dependencies might be nicer if they accepted "dynamic" parameters.

File: templates/workspaces/unrestricted/porter.yaml [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+      - install
+  - name: sp_id
+    type: string
+    applyTo:
+      - install
+
+mixins:
+  - exec
+  - terraform:
+      clientVersion: 1.1.7
```
[the remark this answers is not in the document]
Comment (marrobi) [reply templates/workspaces/unrestricted/porter.yaml:137]: I took this from the latest release, as that's what the Dockerfile is using.

Didn't want to take main as then a change to the base, will break this bundle.

Thoughts?

File: templates/workspaces/unrestricted/porter.yaml [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+      initFile: providers.tf
```
[the remark this answers is not in the document]
Comment (marrobi) [reply templates/workspaces/unrestricted/porter.yaml:138]: Again, from the one in the latest release. Maybe that's why I got into a muddle with providers.tf Porter required it exists in the Docker build context.

[the remark this answers is not in the document]
Comment (tamirkamara) [reply templates/workspaces/unrestricted/porter.yaml:138]: I see. For this case, you should remove that setting.

File: templates/workspaces/unrestricted/porter.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (tamirkamara) [reply templates/workspaces/unrestricted/porter.yaml:137]: ok, let's keep it as is then.

File: templates/workspaces/unrestricted/porter.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (marrobi) [reply templates/workspaces/unrestricted/porter.yaml]: Sorted, that cost me hours, the providers.tf file has gone now too.

File: templates/workspaces/unrestricted/terraform/providers.tf [hunk 3]
```diff
@@ -0,0 +1,16 @@
+terraform {
```
[the remark this answers is not in the document]
Comment (marrobi) [reply templates/workspaces/unrestricted/terraform/providers.tf]: gone

File: templates/workspaces/unrestricted/Dockerfile.tmpl [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+# another location in this file. If you remove that line, the mixins generated content is appended to this file.
+# PORTER_MIXINS
+
+# Use the BUNDLE_DIR build argument to copy files into the bundle
+
+COPY . $BUNDLE_DIR
+
+WORKDIR $BUNDLE_DIR
+
+RUN  curl -o azuretre.tar.gz -L "https://github.com/microsoft/AzureTRE/archive/refs/tags/v${AZURE_TRE_VERSION}.tar.gz" \
```
[the remark this answers is not in the document]
Comment (marrobi) [reply templates/workspaces/unrestricted/Dockerfile.tmpl]: done, again caught up in the providers.tf stuff.

File: templates/workspaces/unrestricted/Dockerfile.tmpl [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  && tar -xzf azuretre.tar.gz "AzureTRE-${AZURE_TRE_VERSION}/templates/workspaces/base"  --strip-components=4 --skip-old-files \
+  && tar -xzf azuretre.tar.gz "AzureTRE-${AZURE_TRE_VERSION}/templates/workspaces/base/terraform/providers.tf" --strip-components=4 --overwrite \
```
[the remark this answers is not in the document]
Comment (marrobi) [reply templates/workspaces/unrestricted/Dockerfile.tmpl]: done, again caught up in the providers.tf stuff.

File: templates/workspaces/unrestricted/Dockerfile.tmpl [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+# You can control where the mixin's Dockerfile lines are inserted into this file by moving "# PORTER_MIXINS" line
+# another location in this file. If you remove that line, the mixins generated content is appended to this file.
+# PORTER_MIXINS
+
+# Use the BUNDLE_DIR build argument to copy files into the bundle
+
+COPY . $BUNDLE_DIR
+
+
```
Comment (tamirkamara) [inline templates/workspaces/unrestricted/Dockerfile.tmpl:35]: I'm missing the lines that enable this to be installed without network access. Check another bundle for an example.

**Review by tamirkamara — APPROVED** [review]

**Pull request merged**